packages/kit/src/packaging/index.js

```diff
@@ -86,19 +82,20 @@ export async function make_package(config, cwd = process.cwd()) {
 
 		write(path.join(cwd, config.kit.package.dir, out_file), out_contents);
 
-		if (!user_defined_exports && exports_filter(file)) {
+		if (exports_filter(file)) {
 			const entry = `./${out_file.replace(/\\/g, '/')}`;
 			const key = entry.endsWith('/index.js') ? entry.slice(0, -'/index.js'.length) : entry;
-			pkg.exports[key] = entry;
+			generated[key] = entry;
 		}
 	}
 
-	const main = pkg.exports['./index.js'] || pkg.exports['./index.svelte'];
-
-	if (!user_defined_exports && main) {
-		pkg.exports['.'] = main;
+	if (!pkg.exports) {
+		pkg.exports = generated;
+	} else if (config.kit.package.exports.behavior === 'replace') {
+		pkg.exports['./package.json'] = './package.json';
```

Comment:
I'm not sure this is correct. If there are exports present already, and mode is `replace`, only the `package.json` is overwritten (at least I read the code like that, maybe something above this `if` happens that changes this). Altering the replace-test (see my other comment) might uncover this.